frontend/src/pages/matching/Matching.tsx

```diff
@@ -12,7 +12,7 @@ enum MatchingStateEnum {
   MATCHED,
 }
 
-const Matching: React.FC = () => {
+const Matching: React.FC<{ userId: number }> = ({ userId }) => {
```

Comment:
```suggestion
const Matching: React.FC = () => {
  const userId = useAppSelector(selectUser)?.id;
```

(this should give the id, so we won't need useContext or passing props)